capstone/src/main/java/com/google/sps/data/ReadData.java

```diff
@@ -111,20 +111,34 @@ public void readFile(FileReader file, String zip) {
         }
 
         if (inStore == null) {
-          entity.setProperty("zipCode", zip);
+          entity.setProperty("zipCode", zipCode);
           entity.setProperty("id", id);
           entity.setProperty("date", date);
-          entity.setProperty("completion", completion);
+          entity.setProperty("completionStatus", completionStatus);
           entity.setProperty("gender", gender);
           entity.setProperty("ageRange", ageRange);
-          entity.setProperty("answerOne", answerOne);
-          entity.setProperty("answerTwo", answerTwo);
-          entity.setProperty("answerThree", answerThree);
+          entity.setProperty("directExperience", directExperience);
+          entity.setProperty("rating", rating);
+          entity.setProperty("text", text);
           entity.setProperty("score", score);
           entity.setProperty("responseTimeOne", responseTimeOne);
           entity.setProperty("responseTimeTwo", responseTimeTwo);
           entity.setProperty("responseTimeThree", responseTimeThree);
 
+          // Set properties pertaining to the precinct the survey maps to
+          String precinctName = MapData.getPopulationPrecinct(zipCode);
+          Precinct precinctData = FeatureData.getPrecinct(precinctName);
+
+          if (precinctData != null) {
+            entity.setProperty(
+                "averageHouseholdIncome",
+                String.format("%.2f", precinctData.getAverageHouseholdIncome()));
```

Comment:
Are you sure you want to set this as a string? Using a number here would help you do comparisons more reliably.

Also, if you choose to store a number, please consider storing an integer instead of a float. Floats are notorious for precision errors. What we normally do at google is store "micro" versions of something. For example, 29.123456 (float) gets stored as 29123456 (int64) because it doesn't risk losing accuracy when stored.

Reply:
When you read it a "micro", you can always convert it back to a float when you need to represent it like so.

Reply:
I think the data gets upcasted to Object in data store. I am not sure how each value is stored, but I know that when I fetch from data store I have to downcast to store the variables as primitive types and as a String object

Reply:
Would it be possible to store the floats as doubles to prevent precision errors?

Reply:
I think both doubles and floats experience the same error. Computers don't store doubles and floats well as explained in the article below.

Example: https://dzone.com/articles/never-use-float-and-double-for-monetary-calculatio

Essentially what this means is if you store x = 12.2, it might turn into x = 12.19999999999999 after storing it. Therefore, basic comparisons like equality fail.

Reply:
I think that storing as a string for now would work best, the numbers I am dealing with are not that precise and I can cast them to a number when I fetch from data store

Reply:
Mostly, I will be using the numpy module to do the conversions